Blazor WebAssembly build: "ImportError: DLL load failed while importing _ctypes: The specified module could not be found."

### Description

We have a Blazor WebAssembly project we're attempting to publish in our CI build machines. The build fails with the following error:

```
Traceback (most recent call last):
    File "C:\__t\dotnet\packs\Microsoft.NET.Runtime.Emscripten.2.0.23.Sdk.win-x64\6.0.0\tools\emscripten\emcc.py", line 23, in <module>
      from tools.toolchain_profiler import ToolchainProfiler
    File "C:\__t\dotnet\packs\Microsoft.NET.Runtime.Emscripten.2.0.23.Sdk.win-x64\6.0.0\tools\emscripten\tools\toolchain_profiler.py", line 18, in <module>
      from tools import response_file
    File "C:\__t\dotnet\packs\Microsoft.NET.Runtime.Emscripten.2.0.23.Sdk.win-x64\6.0.0\tools\emscripten\tools\response_file.py", line 10, in <module>
      from .utils import WINDOWS
    File "C:\__t\dotnet\packs\Microsoft.NET.Runtime.Emscripten.2.0.23.Sdk.win-x64\6.0.0\tools\emscripten\tools\utils.py", line 10, in <module>
      from . import diagnostics
    File "C:\__t\dotnet\packs\Microsoft.NET.Runtime.Emscripten.2.0.23.Sdk.win-x64\6.0.0\tools\emscripten\tools\diagnostics.py", line 9, in <module>
      import ctypes
    File "<frozen zipimport>", line 259, in load_module
    File "ctypes\__init__.py", line 8, in <module>
  ImportError: DLL load failed while importing _ctypes: The specified module could not be found.

C:\__t\dotnet\packs\Microsoft.NET.Runtime.WebAssembly.Sdk\6.0.2\Sdk\WasmApp.Native.targets(429,5): error MSB3073: The command "emcc --version" exited with code 1. [C:\__w\1\s\src\Wasm\Wasm.Api\Wasm.Api.csproj]
```

The build succeeds locally, which makes it hard to provide reproducible steps. 

The CI build will also succeed if we build without AoT enabled _and_ do not install the workloads. However, if we install the workloads, even building without AoT enabled fails.

Included in the zip is the binlog of the failing build:
[logs.zip](https://github.com/dotnet/runtime/files/8135197/logs.zip)


### Reproduction Steps

Like I mentioned above, I haven't been able to find a minimal repro locally - it only reproduces in our CI builds. We are running in a docker image that includes VS2019, but we install dotnet 6.0.100 and use that command to publish the app. The steps in CI are basically:

1. Install dotnet 6.0.100
2. Run `dotnet workload install macos`
3. Run `dotnet workload install wasm-tools`
4. Run `dotnet build <project> -c Release -p:DeployOnBuild=true -p:PublishProfile=<publish profile>`


### Expected behavior

Build succeeds.

### Actual behavior

Build fails with:
```
Traceback (most recent call last):
    File "C:\__t\dotnet\packs\Microsoft.NET.Runtime.Emscripten.2.0.23.Sdk.win-x64\6.0.0\tools\emscripten\emcc.py", line 23, in <module>
      from tools.toolchain_profiler import ToolchainProfiler
    File "C:\__t\dotnet\packs\Microsoft.NET.Runtime.Emscripten.2.0.23.Sdk.win-x64\6.0.0\tools\emscripten\tools\toolchain_profiler.py", line 18, in <module>
      from tools import response_file
    File "C:\__t\dotnet\packs\Microsoft.NET.Runtime.Emscripten.2.0.23.Sdk.win-x64\6.0.0\tools\emscripten\tools\response_file.py", line 10, in <module>
      from .utils import WINDOWS
    File "C:\__t\dotnet\packs\Microsoft.NET.Runtime.Emscripten.2.0.23.Sdk.win-x64\6.0.0\tools\emscripten\tools\utils.py", line 10, in <module>
      from . import diagnostics
    File "C:\__t\dotnet\packs\Microsoft.NET.Runtime.Emscripten.2.0.23.Sdk.win-x64\6.0.0\tools\emscripten\tools\diagnostics.py", line 9, in <module>
      import ctypes
    File "<frozen zipimport>", line 259, in load_module
    File "ctypes\__init__.py", line 8, in <module>
  ImportError: DLL load failed while importing _ctypes: The specified module could not be found.

C:\__t\dotnet\packs\Microsoft.NET.Runtime.WebAssembly.Sdk\6.0.2\Sdk\WasmApp.Native.targets(429,5): error MSB3073: The command "emcc --version" exited with code 1. [C:\__w\1\s\src\Wasm\Wasm.Api\Wasm.Api.csproj]
```

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information
SDK versions:
* Microsoft.NET.Runtime.Emscripten: 2.0.23
* Microsoft.NET.Runtime.WebAssembly.Sdk: 6.0.2

--- a/src/mono/wasm/build/WasmApp.Native.targets
+++ b/src/mono/wasm/build/WasmApp.Native.targets
@@ -256,6 +256,7 @@
 
       <EmscriptenEnvVars Include="EMSDK_PYTHON=$(EmscriptenPythonToolsPath)python.exe" Condition="'$(OS)' == 'Windows_NT'" />
       <EmscriptenEnvVars Include="PYTHONPATH=$(EmscriptenPythonToolsPath)" Condition="'$(OS)' == 'Windows_NT'" />
+      <EmscriptenEnvVars Include="PYTHONHOME=" Condition="'$(OS)' == 'Windows_NT'" />
       <EmscriptenEnvVars Include="EM_CACHE=$(WasmCachePath)" Condition="'$(WasmCachePath)' != ''" />
     </ItemGroup>
 

